Expensify/App
# Update React config to specify dev and production environments

### Details
Mobile doesn't use webPack, so the configuration we were previously using to set local/prod values wasn't working on mobile. This PR updates the config so it works for iOS, Android, and Web. This is done via [this library](https://github.com/tanhauhau/react-web-config). I've added a few new files that will be used for env variables:

`.env` - this is the default dev env values, it is in the gitignore
`.env.production` - this is the release/production values, it is used when we run the mobile deploy script or if we are using the `webpack.production.js` config.

**Note this is being merged into https://github.com/AndrewGable/ReactNativeChat/pull/133**

### Tests
1. Run `npm install` from the root and `pod install` from the iOS dir
2. Run the mobile app via `npm run ios` and `npm run android`, confirm both screens show the `Promise.settled. is not. afunction` error message (being fixed in https://github.com/AndrewGable/ReactNativeChat/pull/133).
3. Run the web app via `npm run web`, confirm that it runs properly and the reports load in the LHN
4. To test the prod vars, run `ENVFILE=.env.production react-native run-ios` and confirm that the reports don't load (its because they don't exist in your env)
